arangod/Agency/Agent.cpp

```diff
@@ -1629,11 +1648,12 @@ void Agent::rebuildDBs() {
   // Apply logs from last applied index to leader's commit index
   LOG_TOPIC("b12cb", DEBUG, Logger::AGENCY)
       << "Rebuilding key-value stores from index " << lastCompactionIndex
-      << " to " << _commitIndex << " " << _state;
+      << " to " << _commitIndex.load(std::memory_order_relaxed) << " " << _state;
 
   {
-    auto logs = _state.slices(lastCompactionIndex + 1, _commitIndex);
-    _readDB.applyLogEntries(logs, _commitIndex, term, false /* do not send callbacks */);
+    auto logs = _state.slices(lastCompactionIndex + 1, _commitIndex.load(std::memory_order_relaxed));
+    _readDB.applyLogEntries(logs, _commitIndex.load(std::memory_order_relaxed),
+                            term, false /* do not send callbacks */);
```

Comment:
We are doing two loads of `_commitIndex` here, which may not return the same value.
There is at least one code location which updates `_commitIndex` without holding any lock:
https://github.com/arangodb/arangodb/pull/12781/files?diff=unified&w=1#diff-a08c67ca9d8698cb38a963fd20dd4999d9163875d86fb777944a888db0222700R436

Reply:
done